### PR TITLE
Added GZIP compression for build context creation

### PR DESCRIPTION
**Pull request by marcuslinke**

### CODE REVIEW

**Comment by mkuzmin** [comment]
This PR is a fix for #283.

File: src/main/java/com/github/dockerjava/core/CompressArchiveUtil.java [hunk 1]
```diff
@@ -16,7 +18,8 @@ public static File archiveTARFiles(File base, Iterable<File> files, String archi
             throws IOException {
         File tarFile = new File(FileUtils.getTempDirectoryPath(), archiveNameWithOutExtension + ".tar");
         tarFile.deleteOnExit();
-        TarArchiveOutputStream tos = new TarArchiveOutputStream(new FileOutputStream(tarFile));
+        TarArchiveOutputStream tos = new TarArchiveOutputStream(new GZIPOutputStream(
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/core/CompressArchiveUtil.java]: i think you need somewhere wrap try with resources

Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/core/CompressArchiveUtil.java]: just wrap as
``` 
try (TarArchiveOutputStream tos = new TarArchiveOutputStream(new GZIPOutputStream..... {
main code
}
``
and remove finally.close, AFAIR tos may become null and then you will have NPEs, try-with-resources should save you.

Comment (marcuslinke) [inline src/main/java/com/github/dockerjava/core/CompressArchiveUtil.java]: Thanks! Will do so.

**Pull request merged**